docs/reference/configuration/storage-config.md

```diff
@@ -250,57 +209,12 @@ Below is the `FILESYSTEM` configuration in `mbed_lib.json`:
 
 If the file system is not set, the default file system and block device will be applied, and `blockdevice`, `external_size` and `external_base_address` will be ignored.
 
-### FILESYSTEM_NO_RBP
-
-<span class="images">![FILESYSTEM](../../images/FILESYSTEM_no_rbp.jpg)<span>`FILESYSTEM_NO_RBP`</span></span>
-
-The `FILESYSTEM_NO_RBP` configuration resembles the `EXTERNAL_NO_RBP` with the difference that it uses FileSystemStore on the external flash. By default, FileSystemStore uses the default file system and the default block device. This configuration has no support for rollback protection and is therefore less secure.
-
-In this configuration, all KVStore C API calls are mapped to the SecureStore class. This class handles the use of the external FileSystemStore.
-
-You can enable this configuration by setting `storage_type` to `FILESYSTEM_NO_RBF` in `storage mbed_lib.json`.
-
-Below is the `FILESYSTEM` configuration in `mbed_lib.json`:
-
-```
-{
-    "name": "storage_filesystem_no_rbp",
-    "config": {
-        "filesystem": {
-            "help": "Options are default, FAT or LITTLE. If default value the filesystem is chosen by the blockdevice type",
-            "value": "default"
-        },
-        "blockdevice": {
-            "help": "Options are default, SPIF, DATAFLASH, QSPIF, SD or other. If default the block device will be chosen by the defined component. If other, override get_other_blockdevice() to support block device out of Mbed OS tree.",
-            "value": "default"
-        },
-        "external_size": {
-            "help": "Size in bytes of the external block device, if default the maximum size available is used.",
-            "value": "0"
-        },
-        "external_base_address": {
-            "help": "The default will set start address to address 0",
-            "value": "0"
-        },    
-        "mount_point": {
-            "help": "Where to mount the filesystem.",
-            "value": "kv"
-        },
-        "folder_path": {
-            "help": "Path for the working directory where the FileSystemStore stores the data",
-            "value": "kvstore"
-        }
-    }
-}
-```
-
-If the file system is not set, the default file system and block device will be applied, and `blockdevice`, `external_size` and `external_base_address` will be ignored.
 
 ### Configuration functions API
 
 Applications must call the function **storage_configuration()** to instantiate the required configuration. This function is defined as weak to allow the replacement of this function with a completely different implementation of the instantiation of components:
 
-```
+```c++
```

Comment:
This failed Travis, because it's not a complete snippet - `_STORAGE_CONFIG` is not declared in the current scope. The documentation in the Mbed OS repo does provide a complete one: https://github.com/ARMmbed/mbed-os/blob/master/storage/docs/Configuration/CONFIGURATION.md#configuration-functions-api

For this PR, maybe we can fix this snippet by copying things over. But in my opinion it's better to avoid duplication - should we have this kind of docs in Mbed OS or here? @Patater

Reply:
@LDong-Arm I'm not sure if this will work but can we include [kv_config.h](https://github.com/ARMmbed/mbed-os/blob/73896715e6202e187ba624fc3853299e73390f19/storage/kvstore/kv_config/include/kv_config/kv_config.h#L29-L30) in the code snippet?

Reply:
Good idea! Each snippet is compiled as an Mbed application, so it might work. Try to push the change and see if Travis passes.

Reply:
I don’t see `storage_configuration()` anywhere in Mbed OS code. The doc says

> Applications must call the function **storage_configuration()** to instantiate the required configuration.

but in fact no such thing is required, because [kvstore_global_api.cpp](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.9.0/storage/kvstore/kvstore_global_api/source/kvstore_global_api.cpp) calls [kv_init_storage_config()](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.9.0/storage/kvstore/kv_config/source/kv_config.cpp#L991-L1018) to initialize storage on every API call.

Maybe `kv_init_storage_config()` is what this doc actually means, but even then applications are not required to call it. I wonder if it makes sense to remove this section.

Reply:
You're right it, I can't see `storage_configuration()` either, its probably an error in the documentation. I might be missing something here but I think they may need to call `kv_init_storage_config()` if they aren't using the API defined in [kvstore_global_api.cpp](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.9.0/storage/kvstore/kvstore_global_api/source/kvstore_global_api.cpp). As I understand it this API works in parallel with KVStore, which the user should still be able to use without it.

Reply:
[`kv_init_storage_config()`](https://github.com/ARMmbed/mbed-os/blob/mbed-os-6.9.0/storage/kvstore/kv_config/source/kv_config.cpp#L991-L1018) sets up _global_ instances of `KVStore`s (via `_storage_config_*()`) and adds those instances to `KVMap` whose purpose is to globally manage multiple `KVStore`s. And the main user of `KVMap` is `kvstore_global_api` which automatically calls `kv_init_storage_config()`. I'm not sure the map is supposed to be used directly by a user, without `kvstore_global_api` - the former is more like a utility just for the latter.

> As I understand it this API works in parallel with KVStore, which the user should still be able to use without it.

When a KVStore (TDBStore, FileSystemStore, SecureStore) is used directly without kvstore_global_api, it's a standalone instances locally managed by the application/user, and the global configurations we are looking at are irrelevant.

But anyway, the change looks okay to me for now - this is not originally the main point of this PR.